processor/processor.go

```diff
@@ -1962,36 +1968,38 @@ func (proc *Handle) processJobsForDestV2(partition string, subJobs subJob) (*tra
 	}
 
-	return proc.generateTransformationMessage(
-		&preTransformationMessage{
-			partition:                    partition,
-			subJobs:                      subJobs,
-			eventSchemaJobs:              eventSchemaJobs,
-			archivalJobs:                 archivalJobs,
-			connectionDetailsMap:         connectionDetailsMap,
-			statusDetailsMap:             statusDetailsMap,
-			reportMetrics:                reportMetrics,
-			destFilterStatusDetailMap:    destFilterStatusDetailMap,
-			inCountMetadataMap:           inCountMetadataMap,
-			inCountMap:                   inCountMap,
-			outCountMap:                  outCountMap,
-			totalEvents:                  totalEvents,
-			marshalStart:                 marshalStart,
-			groupedEventsBySourceId:      groupedEventsBySourceId,
-			eventsByMessageID:            eventsByMessageID,
-			procErrorJobs:                procErrorJobs,
-			jobIDToSpecificDestMapOnly:   jobIDToSpecificDestMapOnly,
-			groupedEvents:                groupedEvents,
-			uniqueMessageIdsBySrcDestKey: uniqueMessageIdsBySrcDestKey,
-			statusList:                   statusList,
-			jobList:                      jobList,
-			start:                        start,
-			sourceDupStats:               sourceDupStats,
-			dedupKeys:                    dedupKeys,
-		})
-}
-
-func (proc *Handle) generateTransformationMessage(preTrans *preTransformationMessage) (*transformationMessage, error) {
+	return &preTransformationMessage{
+		partition:                    partition,
+		subJobs:                      subJobs,
+		eventSchemaJobs:              eventSchemaJobs,
+		archivalJobs:                 archivalJobs,
+		connectionDetailsMap:         connectionDetailsMap,
+		statusDetailsMap:             statusDetailsMap,
+		reportMetrics:                reportMetrics,
+		destFilterStatusDetailMap:    destFilterStatusDetailMap,
+		inCountMetadataMap:           inCountMetadataMap,
+		inCountMap:                   inCountMap,
+		outCountMap:                  outCountMap,
+		totalEvents:                  totalEvents,
+		marshalStart:                 marshalStart,
+		groupedEventsBySourceId:      groupedEventsBySourceId,
+		eventsByMessageID:            eventsByMessageID,
+		procErrorJobs:                procErrorJobs,
+		jobIDToSpecificDestMapOnly:   jobIDToSpecificDestMapOnly,
+		groupedEvents:                groupedEvents,
+		uniqueMessageIdsBySrcDestKey: uniqueMessageIdsBySrcDestKey,
+		statusList:                   statusList,
+		jobList:                      jobList,
+		start:                        start,
+		sourceDupStats:               sourceDupStats,
+		dedupKeys:                    dedupKeys,
+	}
+}
+
+func (proc *Handle) generateTransformationMessage(preTrans *preTransformationMessage) *transformationMessage {
+	if proc.limiter.preTransform != nil {
+		defer proc.limiter.preTransform.BeginWithPriority(preTrans.partition, proc.getLimiterPriority(preTrans.partition))()
```

Comment:
```suggestion
		defer proc.limiter.preTransform.BeginWithPriority("", proc.getLimiterPriority(preTrans.partition))()
```
We can skip passing partition as that field is only used for metrics